csvapi/consumer.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import asyncio
+from csvapi.parseview import ParseView
+from csvapi.profileview import ProfileView
+from csvapi.setup_logger import logger
+import requests
+import os
+
+from config import DB_ROOT_DIR
+
+MINIO_URL = os.environ.get("MINIO_URL", "http://localhost:9000")
+MINIO_USER = os.environ.get("MINIO_USER", "minio")
+MINIO_PASSWORD = os.environ.get("MINIO_PASSWORD", "password")
+
+from csvapi.utils import get_hash, create_connection
+
+import boto3
+from botocore.client import Config, ClientError
+import json
+from datetime import datetime
+import pandas as pd
+
+def run_process_message(key: str, data: dict, topic: str) -> None:
+    asyncio.get_event_loop().run_until_complete(process_message(key, data, topic))
+
+async def process_message(key: str, message: dict, topic: str) -> None:
+    # Get url
+    # Should think if we keep that
+    #r = requests.get('https://www.data.gouv.fr/api/1/datasets/{}/resources/{}'.format(message['meta']['dataset_id'], key))
+    #url = r.json()['url']
+    if((message is not None) & (message['service'] == 'csvdetective')):
+        #try:
+            url = 'https://www.data.gouv.fr/fr/datasets/r/{}'.format(key)
```

Comment:
Why do we need to build an url instead of using the minio location?
This data.gouv.fr location url is environment dependent (dev / demo / prod).